### PR TITLE
doc: adding a note about not conflicting with built-in stuff

### LINKED ISSUE #24459: Keycloak fails to start when uninstalling custom provider
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

core

### Describe the bug

Hi.

I'm working with KeyCloak (_both, with a Docker image and a locally downloaded v22.0.5_) and a custom mapper for OIDC Tokens.

Basically, I've used the code in this URL:
https://betterjavacode.com/programming/custom-protocol-mapper-keycloak

I've generated a JAR and put it into the /opt/keycloak/providers folder. Let's call it: "`provider-cpm_1.0.0.jar`".

Later, I've used it (I can see it as a mapper) in a custom scope within a own Realm's client. It works.
( KeyCloak in a docker container from: keycloak/keycloak:latest )

But, if I want to "uninstall" the mapper from KeyCloak, I just delete the file, as par:
https://www.keycloak.org/server/configuration-provider#_installing_and_uninstalling_a_provider

The problem is that, if I stop the container and start it again, then a nice "`NoSuchFileException`" error message comes:

```
2023-11-01 11:50:37 The DelayedHandler was closed before any children handlers were configured. Messages will be written to stderr.
2023-11-01 11:50:37 2023-11-01 10:50:37,162 DEBUG [org.jboss.logging] (main) Logging Provider: org.jboss.logging.JBossLogManagerProvider
2023-11-01 11:50:37 
2023-11-01 11:50:37 Exception in thread "main" java.lang.reflect.InvocationTargetException
2023-11-01 11:50:37     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2023-11-01 11:50:37     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2023-11-01 11:50:37     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2023-11-01 11:50:37     at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2023-11-01 11:50:37     at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:61)
2023-11-01 11:50:37     at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:32)
2023-11-01 11:50:37 Caused by: java.lang.RuntimeException: Failed to open /opt/keycloak/lib/../providers/provider-cpm_1.0.0.jar
2023-11-01 11:50:37     at io.quarkus.bootstrap.runner.JarResource.ensureJarFileIsOpen(JarResource.java:172)
2023-11-01 11:50:37     at io.quarkus.bootstrap.runner.JarResource.readLockAcquireAndGetJarReference(JarResource.java:153)
2023-11-01 11:50:37     at io.quarkus.bootstrap.runner.JarResource.getResourceURL(JarResource.java:101)
2023-11-01 11:50:37     at io.quarkus.bootstrap.runner.RunnerClassLoader.findResources(RunnerClassLoader.java:214)
2023-11-01 11:50:37     at java.base/java.lang.ClassLoader.getResources(ClassLoader.java:1473)
2023-11-01 11:50:37     at io.smallrye.common.classloader.ClassPathUtils.consumeAsPaths(ClassPathUtils.java:84)
2023-11-01 11:50:37     at io.smallrye.config.AbstractLocationConfigSourceLoader.tryClassPath(AbstractLocationConfigSourceLoader.java:139)
2023-11-01 11:50:37     at io.smallrye.config.AbstractLocationConfigSourceLoader.loadConfigSources(AbstractLocationConfigSourceLoader.java:102)
2023-11-01 11:50:37     at io.smallrye.config.AbstractLocationConfigSourceLoader.loadConfigSources(AbstractLocationConfigSourceLoader.java:85)
2023-11-01 11:50:37     at io.quarkus.runtime.configuration.ApplicationPropertiesConfigSourceLoader$InClassPath.getConfigSources(ApplicationPropertiesConfigSourceLoader.java:30)
2023-11-01 11:50:37     at io.quarkus.runtime.configuration.ApplicationPropertiesConfigSourceLoader$InClassPath.getConfigSources(ApplicationPropertiesConfigSourceLoader.java:27)
2023-11-01 11:50:37     at io.smallrye.config.SmallRyeConfig$ConfigSources.buildSources(SmallRyeConfig.java:581)
2023-11-01 11:50:37     at io.smallrye.config.SmallRyeConfig$ConfigSources.<init>(SmallRyeConfig.java:537)
2023-11-01 11:50:37     at io.smallrye.config.SmallRyeConfig.<init>(SmallRyeConfig.java:69)
2023-11-01 11:50:37     at io.smallrye.config.SmallRyeConfigBuilder.build(SmallRyeConfigBuilder.java:629)
2023-11-01 11:50:37     at io.quarkus.runtime.configuration.QuarkusConfigFactory.getConfigFor(QuarkusConfigFactory.java:29)
2023-11-01 11:50:37     at io.smallrye.config.SmallRyeConfigProviderResolver.getConfig(SmallRyeConfigProviderResolver.java:76)
2023-11-01 11:50:37     at io.smallrye.config.SmallRyeConfigProviderResolver.getConfig(SmallRyeConfigProviderResolver.java:64)
2023-11-01 11:50:37     at org.keycloak.quarkus.runtime.configuration.Configuration.getConfig(Configuration.java:51)
2023-11-01 11:50:37     at org.keycloak.quarkus.runtime.cli.Picocli.hasConfigChanges(Picocli.java:239)
2023-11-01 11:50:37     at org.keycloak.quarkus.runtime.cli.Picocli.requiresReAugmentation(Picocli.java:141)
2023-11-01 11:50:37     at org.keycloak.quarkus.runtime.cli.Picocli.runReAugmentationIfNeeded(Picocli.java:126)
2023-11-01 11:50:37     at org.keycloak.quarkus.runtime.cli.Picocli.parseAndRun(Picocli.java:95)
2023-11-01 11:50:37     at org.keycloak.quarkus.runtime.KeycloakMain.main(KeycloakMain.java:88)
2023-11-01 11:50:37     ... 6 more
2023-11-01 11:50:37 Caused by: java.nio.file.NoSuchFileException: /opt/keycloak/lib/../providers/provider-cpm_1.0.0.jar
2023-11-01 11:50:37     at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
2023-11-01 11:50:37     at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
2023-11-01 11:50:37     at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
2023-11-01 11:50:37     at java.base/sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55)
2023-11-01 11:50:37     at java.base/sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:148)
2023-11-01 11:50:37     at java.base/sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:99)
2023-11-01 11:50:37     at java.base/java.nio.file.Files.readAttributes(Files.java:1851)
2023-11-01 11:50:37     at java.base/java.util.zip.ZipFile$Source.get(ZipFile.java:1394)
2023-11-01 11:50:37     at java.base/java.util.zip.ZipFile$CleanableResource.<init>(ZipFile.java:716)
2023-11-01 11:50:37     at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:250)
2023-11-01 11:50:37     at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:179)
2023-11-01 11:50:37     at java.base/java.util.jar.JarFile.<init>(JarFile.java:346)
2023-11-01 11:50:37     at io.smallrye.common.io.jar.JarFiles.create(JarFiles.java:33)
2023-11-01 11:50:37     at io.quarkus.bootstrap.runner.JarResource.ensureJarFileIsOpen(JarResource.java:170)
2023-11-01 11:50:37     ... 29 more
```

Any idea of how to avoid this situation?

Many thanks in advance.

### Version

22.0.5

### Expected behavior

The "`build`" step, just before "`start | start-dev`" should take care of "non-existing" providers and do not stop server to initiate.
Maybe, throwing an error message in the log file, but not stop it; or providing an alternate way to start or instructions on how to remediate the bug.

### Actual behavior

Breaks the startup process, and the eserver is not starting

### How to Reproduce?

1. Put a JAR file for a mapper in the "`/providers`" folder.
2. Start the server
3. Remove the file
4. Stop the server
5. Restart again

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by shawkins**
closes: #24459

### CODE REVIEW

File: docs/documentation/server_development/topics/providers.adoc [hunk 1]
```diff
@@ -216,10 +216,13 @@ need to be retrieved with the usage of `KeycloakSessionFactory`. It is not recom
 
 === Registering provider implementations
 
-Providers are registered with the server by simply copying them to the `providers` directory.
+Providers are registered with the server by simply copying the JAR file to the `providers` directory.
 
 If your provider needs additional dependencies not already provided by Keycloak copy these to the `providers` directory.
 
+NOTE: Do not include resources or classes in your provider jars that conflict with built-in resources or classes.
```
Comment (vmuzikar) [inline docs/documentation/server_development/topics/providers.adoc:223]: Should we perhaps be more specific here to where the users need to look to and what's considered a built-in resource?

Comment (shawkins) [reply docs/documentation/server_development/topics/providers.adoc:223]: application.properties is called out specifically in the next line. Unfortunately there's not a great way for users to look for overlap on their own. The obvious ones are in our quarkus runtime resources, but there's plenty more if you took a full listing of all the non-class files in all of the built-in jars.

Comment (vmuzikar) [reply docs/documentation/server_development/topics/providers.adoc:223]: Perhaps we could point the users to the `lib/lib/main` dir?

Comment (vmuzikar) [reply docs/documentation/server_development/topics/providers.adoc:223]: I'd to both. Any hint might be a big help to a desperate user. :)

Comment (shawkins) [reply docs/documentation/server_development/topics/providers.adoc]: Ok, expanded on this with more details.

**Review by vmuzikar — APPROVED** [review]
LGTM, thank you @shawkins!

**Pull request merged**